src/components/Select/Select.tsx

```diff
@@ -0,0 +1,130 @@
+import { SelectProps } from './Select.types';
+import React, { useState, useMemo } from 'react';
+import {
+  FormControl,
+  Select as MuiSelect,
+  MenuItem,
+  InputLabel,
+  ListSubheader,
+  TextField,
+  InputAdornment,
+  Divider,
+  Box,
+  CircularProgress,
+} from '@mui/material';
+import SearchIcon from '@mui/icons-material/Search';
+
+/**
+ * This component kets the user select an item from a dropdown.
+ *
+ * @return {JSX.Element}
+ */
+const Select: React.FC<SelectProps> = ({
```

Comment:
This seems to me really specific Select component. I am thinking if we should rename it like 'SelectWithSearch' or 'LanguageSelector'

Reply:
You're right, it's more specific than a standard Select component.